hack/update-codegen.sh

```diff
@@ -18,7 +18,7 @@ CODEGEN_PKG=$(go env GOMODCACHE)/k8s.io/code-generator@v0.27.1
 rm -fr "${APIS:?}/${GROUP:?}"
 rm -fr ./pkg/client
 mkdir -p "${APIS}/${GROUP}/${VERSION}"
-cp -r "${APIS}/${VERSION}/" "${APIS}/${GROUP}/${VERSION}"
+cp -r "${APIS}/${VERSION}"/* "${APIS}/${GROUP}/${VERSION}/"
```

Comment:
please help explain why we made such change here.

Reply:
please change PR title: Let make build rely on make generated to reflect what your PR  does

Reply:
`cp -r "${APIS}/${VERSION}/" "${APIS}/${GROUP}/${VERSION}"`
on linux, it copy the directory to dst regardless if the dst is the same as src, which is false,
on mac,  it copy the files in the directory to dst, which is true.
`cp -r "${APIS}/${VERSION}"/* "${APIS}/${GROUP}/${VERSION}/"`
escape the differences.

Reply:
so, as we talked, 
which is false, you means this operation is incorrect, not we expected, is it？ 

The new change here:
cp -r "${APIS}/${VERSION}"/* "${APIS}/${GROUP}/${VERSION}/"
this will always copy files in the directory to dst and skip difference.

Reply:
Yes.